### PR TITLE
Use configured certificates when connecting to auth proxy in remote access plugin

### PR DESCRIPTION
**Pull request by jarhodes314**
## Proposed changes
This adds support for client certificates to the remote access plugin, so it should successfully connect to the mapper when configured with device certificates.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
- #2802 

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by reubenmiller** [comment]
@jarhodes314 The PR is not set to merge into the `main` branch (hence why the build workflow is not being triggered)

File: crates/common/tedge_config/src/tedge_config_cli/tedge_config.rs [hunk 1]
```diff
@@ -1047,6 +1048,29 @@ pub struct MqttAuthClientConfig {
...
+        .map_err(|e| anyhow!("{e}"))?;
+
+        client_cert_key
+            .map(|(cert, key)| {
+                create_tls_config(
+                    self.ca_path
+                        .or_none()
+                        .map_or(DEFAULT_ROOT_CERT_PATH, |ca| ca.as_str()),
+                    cert,
+                    key,
```
Comment (didier-wenzek) [inline crates/common/tedge_config/src/tedge_config_cli/tedge_config.rs:1066]: `create_tls_config` expects the key first and the cert second.

```suggestion
                    key,
                    cert,
```

**Review by didier-wenzek — COMMENTED** [review]
This is working fine ... once fixed `TEdgeConfigReaderHttp::client_tls_config()` as described bellow  ;-)

**Review by didier-wenzek — APPROVED** [review]
Approved

**Pull request merged**